data-prepper-plugins/avro-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/avro/AvroOutputCodec.java

```diff
@@ -5,51 +5,95 @@
 package org.opensearch.dataprepper.plugins.codec.avro;
 
 import org.apache.avro.Schema;
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumWriter;
 import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
 import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
 import org.opensearch.dataprepper.model.codec.OutputCodec;
 import org.opensearch.dataprepper.model.event.Event;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.io.OutputStream;
-
+import java.util.Objects;
 /**
  * An implementation of {@link OutputCodec} which deserializes Data-Prepper events
  * and writes them to Output Stream as AVRO Data
  */
 @DataPrepperPlugin(name = "avro", pluginType = OutputCodec.class, pluginConfigurationType = AvroOutputCodecConfig.class)
 public class AvroOutputCodec implements OutputCodec {
 
+    private final AvroOutputCodecConfig config;
+    private static final Logger LOG = LoggerFactory.getLogger(AvroOutputCodec.class);
+
     @DataPrepperPluginConstructor
     public AvroOutputCodec(final AvroOutputCodecConfig config) {
-        // TODO: initiate config
+        Objects.requireNonNull(config);
+        this.config = config;
     }
 
+
+    private DataFileWriter<GenericRecord> dataFileWriter;
+
+    private Schema schema;
+
+    private static final String AVRO = "avro";
+
+
     @Override
     public void start(final OutputStream outputStream) throws IOException {
-        // TODO: do the initial wrapping
+        Objects.requireNonNull(outputStream);
+        if(config.getSchema()!=null){
+            schema=parseSchema(config.getSchema());
+        }
+        else if(config.getFileLocation()!=null){
+            schema = AvroSchemaParser.parseSchemaFromJsonFile(config.getFileLocation());
```

Comment:
There is an implicit assumption here that the schema is always provided as a local file. What if the schema is provided as a file in S3?

Reply:
Fetching Schema from a file present in S3 or elsewhere wasn't a part of the initial requirements.